pkg/config/remote/service/service.go

```diff
@@ -172,8 +178,9 @@ func NewService() (*Service, error) {
 		clientsTTL = defaultClientsTTL
 	}
 	clock := clock.New()
+
 	return &Service{
-		ctx:                            context.Background(),
+		ctx:                            ctx,
```

Comment:
I never noticed this before, but I don't think this is how context is intended to be used, as something long lived stored in a struct. This could open up weird behavior if somebody were to replace this with a context that has a deadline, and then tries to reuse it multiple times. Or it gets dirtied with values that persist from request to request. That's not happening here but it'd be easy for it to happen and us not to notice.

I know this isn't a direct change of this PR but it's worth addressing.